scss/_modal.scss

```diff
@@ -19,11 +19,24 @@
   z-index: $zindex-modal;
   display: none;
   overflow: hidden;
+
+  // Vertical align
+  text-align: center;
+  white-space: nowrap;
+
   // Prevent Chrome on Windows from adding a focus outline. For details, see
   // https://github.com/twbs/bootstrap/pull/10951.
   outline: 0;
   -webkit-overflow-scrolling: touch;
 
+  &:before {
+    display: inline-block;
+    height: 100%;
+    margin-right: -0.25em;
```

Comment:
The vertical alignment works by adding in a full height element before the modal dialog so that `vertical-align : center` has something to align against. And because it's `inline-block` you have to account for the spacing between the elements, which is why need a negative margin.